### ISSUE TITLE
On Mac, it repeatedly asks for permission to access removable drive

### ISSUE THREAD

**Issue by JonAbrams** [neil-morrison44/pocket-sync#312]
**Describe the bug**
Since upgrading to v4.12.0, it keeps prompting for permission to read my SD card. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Open Pocket Sync v4.12.0 on MacOS 14.6.1 (Sonoma)

**Expected behaviour**
It asks for permission at most once per session.

**Screenshots**
[See video](https://drive.google.com/file/d/12FD6p7N8RT8nGtKl80FPxUULJn6hcjuo/view?usp=drive_link)

**Logs**
Copy over the log from the folder that opens when you click the "Open Log Folder" button at the bottom of settings.

**Desktop (please complete the following information):**
 - OS: MacOS 14.6.1 (Sonoma)
 - Version v4.12.0

**Additional context**
Every time I hit "yes" a new core shows in the list. Almost like it's asking permission to read each file on my SD card.

**Comment by JonAbrams** [comment]
The problem has now basically gone away. It re-prompts on each load, but only twice now.
<img width="1029" alt="image" src="https://github.com/user-attachments/assets/70b74de5-600c-4a1d-93c2-215aaeef186e">

**Comment by neil-morrison44** [comment]
Yeah that’s annoying, the message is from the OS rather than the app & I think it “learns” that the app is allowed to access the files on the drive eventually - but that resets after every update.

Annoyingly hard to workaround since the in-development version of the app never does it & any local builds will only do it once.

**Comment by JonAbrams** [comment]
Found this: https://forum.makemkv.com/forum/viewtopic.php?p=156277#p156277

Could it be that you need to "deep sign" the package? It's possible that each prompt is triggered by a different library in the package.

**Comment by neil-morrison44** [comment]
Yeah - signing the app would help but that requires ongoing payments to be an Apple developer etc.

I thought it might be one popup per thread or something, but the app does make one initial request for the root folder on the SD card which _should_ be a bottleneck & enable all other requests to go through without having to be manually approved.

It’s an annoying one.

**Comment by JonAbrams** [comment]
The app is useful enough that I can put up with it (so thanks!). I was just worried that the prompts would never go away.
